cachi2/core/utils.py

```diff
@@ -111,3 +112,12 @@ def _copy_using(copy_function: Callable) -> None:
         _copy_using(shutil.copy2)
 
     return destination
+
+
+def get_cache_dir() -> Path:
+    """Return cachi2's global cache directory, useful for storing reusable data."""
+    try:
+        cache_dir = Path(os.environ["XDG_CACHE_HOME"])
+    except KeyError:
+        cache_dir = Path.home().joinpath(".cache")
```

Comment:
Depends :grin: 

It looks like `HOME` gets set according to the user id in the container. By default, the containers run as root, in which case `HOME=/root`. But the end user can override that: https://tekton.dev/docs/pipelines/taskruns/#example-of-running-step-containers-as-a-non-root-user. In which case `HOME=/` and isn't writable for the current user

Reply:
But as long as ~/.cache/cachi2 isn't intended for anything but local usage, this will be fine

If you wanted to future-proof it, https://github.com/containerbuildsystem/cachi2/pull/427#discussion_r1457404703 might work